### ISSUE TITLE
Decimal and float conversions should use invariant culture (unit tests fail)

### ISSUE THREAD

**Issue by schlaup** [graphql-dotnet/graphql-dotnet#302]
First, very nice work!!!

I found one issue:

When the OS is not "en-US" float and decimal values are converted incorrectly.

For example:

```
Failed: GraphQL.Tests.Types.DecimalGraphTypeTests.coerces_numeric_string_to_decimal
  type.ParseValue("12345.6579")
    should be
12345,6579m
    but was
123456579m
```

or

```
Failed: GraphQL.Tests.Bugs.Bug138DecimalPrecisionTests.double_to_decimal_does_not_lose_precision
  writtenResult
    should be
"{
  "data": {
    "set": 24.15
  }
}"
    but was
"{
  "data": {
    "set": 2415.0
  }
}"
```

To reproduce set numeric format of Windows to "German".

or set culture in the test. For example:

```
        [Fact]
        public void coerces_numeric_string_to_decimal()
        {
            System.Threading.Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("de-DE");
            type.ParseValue("12345.6579").ShouldBe((decimal)12345.6579);
        }
```

(Better in an XUnit initializer for all tests. Or something like https://coding.abel.nu/2014/02/rerunning-all-unit-tests-with-different-ui-culture/)

The fix is not difficult. Use InvariantCulture when parsing decimal or floats, for example:

```
        public override object ParseValue(object value)
        {
            decimal result;
            if (decimal.TryParse(value?.ToString() ?? string.Empty, out result))
            {
                return result;
            }
            return null;
        }
```

should be:

```
        public override object ParseValue(object value)
        {
            decimal result;
            if (decimal.TryParse(value?.ToString() ?? string.Empty, NumberStyles.Any, CultureInfo.InvariantCulture, out result))
            {
                return result;
            }
            return null;
        }
```

**Comment by joemcbride** [comment]
👍  Mind doing a PR to fix this?

**Comment by schlaup** [comment]
Ok, the fix is easy but I am not so familiar with xUnit.
How would you want to setup additional tests?

Something like: ?
https://coding.abel.nu/2014/02/rerunning-all-unit-tests-with-different-ui-culture/

The alternative would be to write some additional tests that only cover the tests that are failing now.

**Comment by joemcbride** [comment]
I'm not sure of a good way to run all tests under a specific culture, especially in the .NET Core world.  I would be fine just writing some tests for this specific issue for now.

[Here are some sample tests](https://github.com/xunit/samples.xunit/blob/master/UseCulture/UseCultureAttributeTests.cs) that use `UseCulture` and `Theory` + `InlineData` to run tests with a specific culture.

**Comment by mule** [comment]
Uh, why was this closed, the issue still remains AFAIK?

**Comment by joerneu** [comment]
@schlaup and @mule - The fix is not as easy as described in this issue. There are a couple more places to change (depending if the value is an argument, variable or part of an object). Sometimes a converted numeric value gets converted back to a string and then to a numeric value again. (So the ToString() call needs to use InvariantCulture as well.)

I went ahead and fixed the issue. All tests run successfully and I tested on a different locale. Attached is a GIT patch file for anyone who needs it now or want to look at the required changes.

@joemcbride I cannot provide a PR yet as my fork and IDE is not up to date.

[patch.diff.txt](https://github.com/graphql-dotnet/graphql-dotnet/files/1230907/patch.diff.txt)
